### DISCUSSION TITLE
Own docker image.

### DISCUSSION THREAD

**Discussion by er2de2** [glanceapp/glance#53] [Help] [answered]
I would like to create my own docker image. Which files do I need to edit to make it point to my repository?

**Comment by svilenmarkov** [comment, marked as answer]
Hey,

I'm assuming you're only interested in building the image for a single platform and architecture, in which case it should be just about editing the Dockerfile to:

```dockerfile
FROM alpine:3.19

WORKDIR /app
COPY build/glance /app/glance

EXPOSE 8080/tcp
ENTRYPOINT ["/app/glance"]
```

You can build Glance with:

```
go build -o build/glance .
```

And then build the image with:

```
docker build -t er2de2/glance:latest .
```

The tag, specified with the `-t` option is what determines the name of your image. You can then push that image to your repository with:

```
docker push er2de2/glance:latest
```

**Reply by er2de2** [reply]
Thank you for the clarification

**Comment by er2de2** [comment]
I create my own docker image , but when i try to deploy on my server terminal show me:
exec /app/glance: no such file or directory

**Reply by svilenmarkov** [reply]
What do you get if you list the contents of the directory with this command?

```
docker run --rm --entrypoint /bin/ls er2de2/glance -la /app
```

**Reply by er2de2** [reply]
docker run --rm --entrypoint /bin/ls romandrajer/glance:2.0 -la /app
total 16068
drwxr-xr-x    1 root     root          4096 May 11 08:08 .
drwxr-xr-x    1 root     root          4096 May 11 09:15 ..
-rwxr-xr-x    1 root     root      16442988 May 11 08:07 glance

**Reply by svilenmarkov** [reply]
Ah, right - I forgot about CGO, my bad! Try building Glance with this command and then repeat building the image and then pushing with the previous commands:

```
CGO_ENABLED=0 go build -o build/glance .
```

**Reply by er2de2** [reply]
Now everything work perfect. Thank you very much.
